Libraries/LibChess/Chess.cpp

```diff
@@ -452,12 +533,14 @@ Chess::Result Chess::game_result() const
     return Result::StaleMate;
 }
 
-bool Chess::is_promotion_move(const Move& move, Colour colour) const
+bool Board::is_promotion_move(const Move& move, Colour colour) const
 {
     if (colour == Colour::None)
         colour = turn();
 
-    if (!is_legal(move, colour))
+    Move queen_move = move;
+    queen_move.promote_to = Type::Queen;
+    if (!is_legal(queen_move, colour))
         return false;
```

Comment:
Not sure about this one, but it feels like a legal knight promotion would not be accepted by this algorithm.

Reply:
`is_promotion_move` checks if a pawn is being promoted regardless of what it's being promoted to (even if the promoted piece is invalid). The move `h8=N` or `h8` would become `h8=Q`, only for the purpose of checking that the pawn is in promotion position. This method is used by `ChessWidget` to spawn a `PromotionDialog` to select the type of piece to promote to.